lib/kubeclient/common.rb

```diff
@@ -173,7 +173,7 @@ def create_entity(entity_type, entity_config, klass)
       end
 
       def update_entity(entity_type, entity_config)
-        name = entity_config.name
+        name = entity_config.name || entity_config.metadata.name
```

Comment:
no need to do the "or" part. it's always now part of metadata. v1beta1 is not supported anyway...
so you can just replace the old statement with the new one.